_enumerable.rb

```diff
@@ -0,0 +1,84 @@
+module Enumerable
+  def my_each
+    each do |ele|
+      yield ele
+    end
+    self
+  end
+
+  def my_each_with_index
+    length.times do |indx|
+      yield self[indx], indx
+    end
+    self
+  end
+
+  def my_select
+    selected_elements = []
+    my_each do |k|
+      selected_elements.push(k) if yield k
+    end
+    selected_elements
+  end
+
+  def my_all?
+    my_each do |ele|
+      # returns false if an element that doesn't satsify the condition is found
+      return false unless yield ele
+    end
+    true
+  end
+
+  def my_any?
+    my_each do |ele|
+      return true if yield ele
+    end
+    false
+  end
+
+  def my_none?
+    my_each do |ele|
+      return false if yield ele
+    end
+    true
+  end
+
+  def my_count
+    count = 0
+    my_each do |ele|
+      count += 1 if yield ele
+    end
+    count
+  end
```

Comment:
Is expected that:

#my_count returns the number of items in enum through enumeration

#my_count returns the number of items in enum through enumeration::range

#my_count counts the number of items in enum that are equal to item if an argument is given